### ISSUE TITLE
docker-compose missing a large number of requirements for expected results

### ISSUE THREAD

**Issue by dskvr** [Significant-Gravitas/Auto-GPT#2675]
### ⚠️ Search for existing issues first ⚠️

- [X] I have searched the existing issues, and there is no existing issue for my problem

### Which Operating System are you using?

Docker

### GPT-3 or GPT-4?

GPT-4

### Steps to reproduce 🕹

- run docker-compose
- end process
- no output files are stored locally after closing process, docker crashes, etc

### Current behavior 😯

- volumes that are required to make use of `Auto-GPT` in the majority of use cases are not mounted
- Auto-GPT cannot connect to redis via localhost 

### Expected behavior 🤔

- `docker-compose.yaml` has all the required members to use autogpt in docker. 

```
# To boot the app run the following:
# docker-compose run auto-gpt
version: "3.9"

services:
  auto-gpt:
    depends_on:
      - redis
    build: ./
    env_file:
      - .env
    volumes:
      - "./autogpt:/app"
      - ".env:/app/.env"
      - "./ai_settings.yaml:/ai_settings.yaml"
      - "./auto_gpt_workspace:/home/appuser/auto_gpt_workspace:rw"
    links:
      - "redis"
    profiles: ["exclude-from-up"]

  redis:
    image: "redis/redis-stack-server:latest"
    ports: 
      - "6379:6379"
    volumes: 
      - "/path/to/backup/:/data:rw"
      - "./redis-stack.conf:/redis-stack.conf:ro"
```

in `.env` change `REDIS_HOST` to `redis` 
Add an `ai_settings.yaml` if that's your style (otherwise the user prompt still works)
add a `redis-stack.conf` if you want to be able to persist redis data (you must also change `WIPE_REDIS_ON_START` to `False` ofc)

Example `redis-stack.conf`
```
save 120 1 
dbfilename dump.rdb
```

### Your prompt 📝

not relevant

### Your Logs 📒

not relevant

### Additional 
Would be willing to open a PR, but with how fast things are moving rn I am afraid I do not have the time to both catch up on developments and contribute meaningful changes.

**Comment by caleb-s-cullen** [comment]
I see that `ai_settings.yaml` is mounted on the container from my regular file system, but I still get prompted as if there were no such file anyway.  Docker shows that the file should be present.  I find this confusing, but I am not that familiar with Docker.

**Comment by bradiosd** [comment]
Thank you for this. I was struggling to debug my docker compose file. All of the above works on my Ubuntu server now.

**Comment by Joshfindit** [comment]
Having an issue with this.

I found early that `MEMORY_BACKEND` and `REDIS_HOST` needed to be set to `redis`, and I got AutoGPT running successfully (it doesn't complain or crash, and shows `Using memory of type:  RedisMemory` during responses)

But, it also doesn't seem to remember between sessions.
`WIPE_REDIS_ON_START` is `False`.

I've now added `- "./redis-stack.conf:/redis-stack.conf:ro"` and set the contents of the `redis-stack.conf` file as above (it does generate a dump that looks valid) but it still does not resume.
